packages/cli/src/utils/auth-files-codegen.js

```diff
@@ -45,84 +58,101 @@ const authFileExport = (
+    ),
+    test,
+    comment(
+      `This template string can access all the data returned from the auth test. If you return the test object, you'll access the returned data with a label like \`{{json.X}}\`. If you return \`response.data\` from your test, then your label can be \`{{X}}\`. This can also be a function that returns a label. That function has the standard args \`(${standardArgs(
+        language,
+      ).join(
+        ', ',
+      )})\` and data returned from the test can be accessed in \`bundle.inputData.X\`.`,
+    ),
+    objProperty('connectionLabel', connectionLabel),
+  ];
+
+  const configObj =
+    language === 'typescript'
+      ? objTS('Authentication', ...configProps)
+      : obj(...configProps);
+
   return exportStatement(
     obj(
-      objProperty(
-        'config',
-        obj(
-          comment(authDescription),
-          objProperty('type', strLiteral(authType)),
-          ...extraConfigProps,
-          comment(
-            "Define any input app's auth requires here. The user will be prompted to enter this info when they connect their account.",
-          ),
-          objProperty('fields', arr(...authFields)),
-          comment(
-            "The test method allows Zapier to verify that the credentials a user provides are valid. We'll execute this method whenever a user connects their account for the first time.",
-          ),
-          test,
-          comment(
-            `This template string can access all the data returned from the auth test. If you return the test object, you'll access the returned data with a label like \`{{json.X}}\`. If you return \`response.data\` from your test, then your label can be \`{{X}}\`. This can also be a function that returns a label. That function has the standard args \`(${standardArgs.join(
-              ', ',
-            )})\` and data returned from the test can be accessed in \`bundle.inputData.X\`.`,
-          ),
-          objProperty('connectionLabel', connectionLabel),
-        ),
-      ),
+      objProperty('config', configObj),
       objProperty('befores', arr(...beforeFuncNames)),
       objProperty('afters', arr(...afterFuncNames)),
     ),
+    language,
   );
 };
 
-const authTestFunc = (testUrl = strLiteral(authJsonUrl('me'))) =>
+const authTestFunc = (language, testUrl = strLiteral(authJsonUrl('me'))) =>
   block(
     comment(
       'You want to make a request to an endpoint that is either specifically designed to test auth, or one that every user will have access to. eg: `/me`.',
     ),
     comment(
       'By returning the entire request object, you have access to the request and response data for testing purposes. Your connection label can access any data from the returned response using the `json.` prefix. eg: `{{json.username}}`.',
     ),
-    fatArrowReturnFunctionDeclaration('test', standardArgs, zRequest(testUrl)),
+    fatArrowReturnFunctionDeclaration(
+      'test',
+      standardArgs(language),
```

Comment:
`standardArgs(language)` is the actual change here, since `standardArgs` has been changed from an array to a function.